CONTRIBUTING.md

```diff
@@ -83,19 +76,29 @@ if you choose to install `nextest`.
 
 ### Development
 
-After cloning the repository, run Ruff locally from the repository root with:
+After cloning the repository, Ruff can be run locally from the repository root on a file such as `add_rule.py` using:
 
 ```shell
-cargo run -p ruff -- check /path/to/file.py --no-cache
+cargo run -p ruff -- check ./scripts/add_rule.py --no-cache
 ```
 
-Prior to opening a pull request, ensure that your code has been auto-formatted,
-and that it passes both the lint and test validation checks:
+Now that the repository is cloned, you can optionally install pre-commit hooks to automatically run the validation checks
+when making a commit:
 
 ```shell
-cargo clippy --workspace --all-targets --all-features -- -D warnings  # Rust linting
-RUFF_UPDATE_SCHEMA=1 cargo test  # Rust testing and updating ruff.schema.json
-pre-commit run --all-files --show-diff-on-failure  # Rust and Python formatting, Markdown and Python linting, etc.
+pre-commit install
+```
+
+Prior to opening a pull request you can run the below to ensure that the following has been run respectively:
+
+- Linting against Rust code
+- Rust tests and updating of ruff.schema.json
+- Rust and Python formatting, Markdown and Python linting, etc.
+
+```shell
+cargo clippy --workspace --all-targets --all-features -- -D warnings
+RUFF_UPDATE_SCHEMA=1 cargo test
+pre-commit run --all-files --show-diff-on-failure
```

Comment:
ideally this would be in something like `cargo pre-test` alias, similar to `npm run` scripts, but this doesn't seem to natively exist in cargo